content/1_docs/3_reference/6_system/1_options/0_kirbytext/cheatsheet-article.txt

```diff
@@ -26,3 +26,23 @@ The default height that is added to Youtube and Vimeo iframes (default: `false`
 
 ### `width`
 The default width that is added to Youtube and Vimeo iframes (default: `false` = no width)
+
+## Drag texts
+
+<since v="3.4.3">
+You can define custom callback functions (`pageDragText` or `fileDragText`) for the text that gets inserted when dragging a page or a file on a textarea, e.g.
+
+```php
+return [
+  'kirbytext' => [
+    'fileDragText' => function (Kirby\Cms\File $file, string $url) {
+        if ($file->extension() === 'heic') {
```

Comment:
I must admit I don't understand what this feature is supposed to do and when I add this code in my config, it doesn't seem to do anything.

Reply:
Kirby doesn't recognize the `heic` image file type and doesn't apply the image tag `(image: file.heic)` when the file with heic extension is moved to the textarea field. This feature is useful for customizing the tag during drag and drop.

Reply:
Background: https://github.com/getkirby/kirby/pull/2735

Reply:
Ok, I finally managed to get it to work, but not inside the kirbytext option, but inside the Panel option:

```
'panel' => [
        'kirbytext' => [
            'fileDragText' => function(\Kirby\Cms\File $file, $url) {
                if($file->extension() === 'jpg') {
                    return sprintf('(screenshot: %s)', $url);
                }
        
                if($file->type() === 'video') {
                    return sprintf('(video: %s loop:true)', $url);
                }
        
                return null;
            },
            'pageDragText' => function (\Kirby\Cms\Page $page) {
                return sprintf('Links sind toll: %s', $page->url());
            },
            
        ],
    ],
```
So this needs to be moved.

Reply:
Oh my, that was stupid of me. Sorry!

Reply:
Moved them